### PR TITLE
[pytorch][inference][build] PyTorch 1.13.0 Inference EC2/SM

### PR DESCRIPTION
**Pull request by xncqr**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Build PyTorch 1.13.0 Inference images for EC2 and SageMaker

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [x] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [x] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [x] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [x] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [x] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/dlc_tests/sanity/test_pre_release.py [hunk 1]
```diff
@@ -407,6 +407,9 @@ def _run_dependency_check_test(image, ec2_connection):
         "CVE-2016-6303",
         "CVE-2016-2182",
         "CVE-2022-2068",
+        # Latest installed openssl on ubuntu 20.04 is 1.1.1f-1ubuntu2.16 which is higher than these CVEs. False Positive
+        "CVE-2022-1292",
```
Comment (shantanutrip) [inline test/dlc_tests/sanity/test_pre_release.py:411]: In general we have been in the practice of ignoring these CVEs by adding skips [here](https://github.com/aws/deep-learning-containers/pull/2367/files#diff-8c9ca2c09d29a6f9e48bf73c2a6e606318fb14fa55326b928964dd47c344c0ebR422) and [here](https://github.com/aws/deep-learning-containers/pull/2367/files#diff-8c9ca2c09d29a6f9e48bf73c2a6e606318fb14fa55326b928964dd47c344c0ebR454). However, that was done since, in the past, we had a few images that installed Openssl from source and for those images, these vulns were not false positives. It is a good solution to add these CVEs as a part of this list. However, before doing that, we must ensure that none of the images (training or inference) install Openssl from source.

File: dlc_developer_config.toml [hunk 2]
```diff
@@ -41,14 +41,14 @@ ec2_tests = true
...
-sagemaker_local_tests = false
+sagemaker_local_tests = true
 
 # SM remote test valid values:
 # "off" --> do not trigger sagemaker remote tests (default)
 # "standard" --> run standard sagemaker remote tests from test/sagemaker_tests
 # "rc" --> run release_candidate_integration tests
 # "efa" --> run efa sagemaker tests
-sagemaker_remote_tests = "off"
+sagemaker_remote_tests = "standard"
```
Comment (shantanutrip) [inline dlc_developer_config.toml:51]: Did we run the efa and the performance tests?

File: test/dlc_tests/sanity/test_pre_release.py (hunk 1 above)
Comment (xncqr) [reply test/dlc_tests/sanity/test_pre_release.py]: See they were already there and updated in that location.

File: dlc_developer_config.toml (hunk 2 above)
Comment (xncqr) [reply dlc_developer_config.toml]: both are happening on b6aabfd.

File: test/test_utils/__init__.py [hunk 3]
```diff
@@ -52,8 +52,9 @@ def get_ami_id_ssm(region_name, parameter_path):
     ami_id = eval(ami['Parameter']['Value'])['image_id']
     return ami_id
 
-UBUNTU_18_BASE_DLAMI_US_WEST_2 = get_ami_id_boto3(region_name="us-west-2", ami_name_pattern="Deep Learning AMI GPU CUDA 11.1.1 (Ubuntu 18.04) ????????")
-UBUNTU_18_BASE_DLAMI_US_EAST_1 = get_ami_id_boto3(region_name="us-east-1", ami_name_pattern="Deep Learning AMI GPU CUDA 11.1.1 (Ubuntu 18.04) ????????")
+# Temporary AMIs till the official realease
+UBUNTU_18_BASE_DLAMI_US_WEST_2 = get_ami_id_boto3(region_name="us-west-2", ami_name_pattern="DDeep Learning Base AMI (Ubuntu 18.04) Version ??.?")
```
Comment (arjkesh) [inline test/test_utils/__init__.py:56]: @xncqr appears to be a typo here "DDeep" which is causing EC2 tests to fail

Comment (xncqr) [reply test/test_utils/__init__.py]: just saw that and updated.. ty

File: pytorch/inference/docker/1.13/py3/Dockerfile.cpu [hunk 4]
```diff
@@ -0,0 +1,248 @@
+ARG PYTHON=python3
+ARG PYTHON_VERSION=3.9.13
+ARG MAMBA_VERSION=4.12.0-2
+ARG OPEN_MPI_VERSION=4.1.1
```
Comment (arjkesh) [inline pytorch/inference/docker/1.13/py3/Dockerfile.cpu:4]: What is the reason to use 4.1.1 instead of latest stable version (4.1.4)?

File: pytorch/inference/docker/1.13/py3/cu117/Dockerfile.gpu [hunk 5]
```diff
@@ -0,0 +1,315 @@
+ARG PYTHON=python3
+ARG PYTHON_VERSION=3.9.13
+ARG MAMBA_VERSION=22.9.0-1
+ARG OPEN_MPI_VERSION=4.1.1
```
Comment (arjkesh) [inline pytorch/inference/docker/1.13/py3/cu117/Dockerfile.gpu:4]: What is the reason to use 4.1.1 instead of the latest stable version (4.1.4)?

File: pytorch/buildspec.yml [hunk 6]
```diff
@@ -46,134 +46,119 @@ context:
...
-  BuildSageMakerCPUPTTrainPy3DockerImage:
-    <<: *TRAINING_REPOSITORY
-    build: &PYTORCH_CPU_TRAINING_PY3 false
-    image_size_baseline: 5000
+      <<: *INFERENCE_CONTEXT
+  BuildSageMakerCPUPTInferencePy3DockerImage:
+    <<: *INFERENCE_REPOSITORY
+    build: &PYTORCH_CPU_INFERENCE_PY3 false
+    image_size_baseline: 4899
+    #base_image_name: BuildEC2CPUPTInferencePy3DockerImage
```
Comment (arjkesh) [inline pytorch/buildspec.yml:139]: this can be removed, these are multistage dockerfiles so they don't require this

File: pytorch/buildspec.yml [hunk 7]
```diff
@@ -46,134 +46,119 @@ context:
...
-      <<: *TRAINING_CONTEXT
-  BuildSageMakerGPUPTTrainPy3DockerImage:
-    <<: *TRAINING_REPOSITORY
-    build: &PYTORCH_GPU_TRAINING_PY3 false
+      <<: *INFERENCE_CONTEXT
+  BuildSageMakerGPUPTInferencePy3DockerImage:
+    <<: *INFERENCE_REPOSITORY
+    build: &PYTORCH_GPU_INFERENCE_PY3 false
     image_size_baseline: 14000
+    #base_image_name: BuildEC2GPUPTInferencePy3DockerImage
```
Comment (arjkesh) [inline pytorch/buildspec.yml:153]: this can be removed, these are multistage dockerfiles so they don't require this

File: pytorch/inference/docker/1.13/py3/Dockerfile.cpu [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+ARG PYTHON
+
+ENV SAGEMAKER_SERVING_MODULE sagemaker_pytorch_serving_container.serving:main
+
+# Install scikit-learn and pandas
+RUN conda install -y -c conda-forge \
+    scikit-learn \
+    pandas
+
+RUN pip install --no-cache-dir "sagemaker-pytorch-inference>=2"
```
Comment (arjkesh) [inline pytorch/inference/docker/1.13/py3/Dockerfile.cpu:228]: checking (not a blocker for EC2) - is the latest version of sagemaker-pytorch-inference compatible with TS 0.6.0?

File: pytorch/inference/docker/1.13/py3/cu117/Dockerfile.gpu [hunk 9]
```diff
@@ -0,0 +1,315 @@
...
+ARG PYTHON
+
+ENV SAGEMAKER_SERVING_MODULE sagemaker_pytorch_serving_container.serving:main
+
+# Install scikit-learn and pandas
+RUN conda install -y -c conda-forge \
+    scikit-learn \
+    pandas
+
+RUN pip install --no-cache-dir "sagemaker-pytorch-inference>=2"
```
Comment (arjkesh) [inline pytorch/inference/docker/1.13/py3/cu117/Dockerfile.gpu:295]: checking (not a blocker for EC2) - is the latest version of sagemaker-pytorch-inference compatible with TS 0.6.0?

File: test/test_utils/__init__.py [hunk 10]
```diff
@@ -52,8 +52,9 @@ def get_ami_id_ssm(region_name, parameter_path):
     ami_id = eval(ami['Parameter']['Value'])['image_id']
     return ami_id
 
-UBUNTU_18_BASE_DLAMI_US_WEST_2 = get_ami_id_boto3(region_name="us-west-2", ami_name_pattern="Deep Learning AMI GPU CUDA 11.1.1 (Ubuntu 18.04) ????????")
-UBUNTU_18_BASE_DLAMI_US_EAST_1 = get_ami_id_boto3(region_name="us-east-1", ami_name_pattern="Deep Learning AMI GPU CUDA 11.1.1 (Ubuntu 18.04) ????????")
+# Temporary AMIs till the official realease
```
Comment (arjkesh) [inline test/test_utils/__init__.py:55]: Are we ready to go with the AMIs? Can this comment be removed?

File: pytorch/inference/docker/1.13/py3/Dockerfile.cpu (hunk 8 above)
Comment (xncqr) [reply pytorch/inference/docker/1.13/py3/Dockerfile.cpu:228]: TS 0.6.0  has been around for a while, and the inference toolkit is compatible with it. The Concern was whether TS 0.6.0 was compatible with PT 1.13, which was tested and verified good.

File: test/test_utils/__init__.py (hunk 10 above)
Comment (xncqr) [reply test/test_utils/__init__.py]: Rmoved

File: pytorch/inference/docker/1.13/py3/cu117/Dockerfile.gpu [hunk 11]
```diff
@@ -0,0 +1,315 @@
...
+#                                         |___/
+#  ____           _
+# |  _ \ ___  ___(_)_ __   ___ 
+# | |_) / _ \/ __| | '_ \ / _ \
+# |  _ <  __/ (__| | |_) |  __/
+# |_| \_\___|\___|_| .__/ \___|
+#                  |_|
+########################################################
+
+# FROM nvidia/cuda:11.6.2-base-ubuntu20.04 AS base_image
```
[the remark this answers is not in the document]
Comment (xncqr) [reply pytorch/inference/docker/1.13/py3/cu117/Dockerfile.gpu]: Removed

File: pytorch/inference/docker/1.13/py3/cu117/Dockerfile.gpu (hunk 5 above)
Comment (xncqr) [reply pytorch/inference/docker/1.13/py3/cu117/Dockerfile.gpu]: didn't catch the newer patch.. updated.

File: pytorch/buildspec.yml (hunk 7 above)
Comment (xncqr) [reply pytorch/buildspec.yml]: removed in both places.

File: pytorch/buildspec.yml (hunk 6 above)
Comment (xncqr) [reply pytorch/buildspec.yml]: removed.

File: pytorch/inference/docker/1.13/py3/cu117/Dockerfile.gpu [hunk 12]
```diff
@@ -0,0 +1,314 @@
+ARG PYTHON=python3
+ARG PYTHON_VERSION=3.9.13
+ARG MAMBA_VERSION=22.9.0-1
```
Comment (arjkesh) [inline pytorch/inference/docker/1.13/py3/cu117/Dockerfile.gpu:3]: why do Mamba versions differ btwn CPU and GPU? 

Looks like we've been using 4.12 for a long time, @RadhikaB-97 or @jeet4320 do you see any reason to update/not-update mamba version?

Comment (xncqr) [reply pytorch/inference/docker/1.13/py3/cu117/Dockerfile.gpu:3]: I Updated CPU to match GPU. Was so focused on insuring the GPU was up to date and working, guess I missed that on on the GPU. This would be a good example where the common items between the CPU and GPU images could be a "base".
